testsupport/testcontainers/src/main/java/com/adobe/testing/s3mock/testcontainers/S3MockContainer.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright 2017-2021 Adobe.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.adobe.testing.s3mock.testcontainers;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+
+public class S3MockContainer extends GenericContainer<S3MockContainer> {
+  public static final String IMAGE_NAME = "adobe/s3mock";
+
+  private static final int S3MOCK_DEFAULT_HTTP_PORT = 9090;
+  private static final int S3MOCK_DEFAULT_HTTPS_PORT = 9191;
+  private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse(IMAGE_NAME);
+
+  /**
+   * Create a S3MockContainer.
+   *
+   * @param tag in the format of "2.1.27"
+   */
+  public S3MockContainer(String tag) {
+    this(DEFAULT_IMAGE_NAME.withTag(tag));
+  }
+
+  /**
+   * Create a S3MockContainer.
+   *
+   * @param dockerImageName in the format of {@link DockerImageName#parse(String)} where the
+   *                        parameter is the full image name like "adobe/s3mock:2.1.27"
+   */
+  public S3MockContainer(final DockerImageName dockerImageName) {
+    super(dockerImageName);
+
+    dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+    waitingFor(Wait.forHttp("/favicon.ico")
+        .forPort(S3MOCK_DEFAULT_HTTP_PORT)
+        .withMethod("GET")
+        .forStatusCode(200));
+    addExposedPort(S3MOCK_DEFAULT_HTTP_PORT);
+    addExposedPort(S3MOCK_DEFAULT_HTTPS_PORT);
+  }
+
+  public S3MockContainer withValidKmsKeys(String kmsKeys) {
+    this.addEnv("validKmsKeys", kmsKeys);
+    return self();
+  }
+
+  public S3MockContainer withInitialBuckets(String initialBuckets) {
+    this.addEnv("initialBuckets", initialBuckets);
+    return self();
+  }
+
+  public S3MockContainer withRoot(String root) {
+    this.addEnv("root", root);
```

Comment:
shouldn't this also require mounting a related volume?

Reply:
of course, you're right - this did not make sense. We're now accepting an absolute path on the host system and mount that as the "root".